### ISSUE TITLE
`docker-healthcheck.service` not started on debian default image

### ISSUE THREAD

**Issue by robinmonjo** [kubernetes/kops#2279]
Hello all,

I have seen a lot of docker daemon becoming unresponsive in my cluster. Turns out that the debian image provides `docker-healthcheck.timer` and `docker-healthcheck.service` to make sure the docker daemon is running and restart it if needed. However it is never started and so useless unless I run `systemctl start docker-healthcheck.service`. This activate the timer.

Is this a bug ? Any reason this is not started by default ?

**Comment by netflash** [comment]
Same story here, I also executed `systemctl enable docker-healthcheck.service` to make sure it will work after reboot.

**Comment by netflash** [comment]
BTW I got to this script from issue #2178

**Comment by bcorijn** [comment]
I'm having some docker hangups of which my machines do not recover without manual restart, output from sysctl:
```
UNIT                           LOAD   ACTIVE SUB       DESCRIPTION
docker-healthcheck.timer       loaded active elapsed   Trigger docker-healthcheck periodically
```

Using Kubernetes 1.5.4, Kops 1.5.3 and AMI `kope.io/k8s-1.5-debian-jessie-amd64-hvm-ebs-2017-01-09`

**Comment by jasonbrownbridge** [comment]
In case this helps anyone you can do something like this to start the service on all nodes in cluster. I used this approach as I think it is more robust to changes in auto scaling group or rolling updates:
```bash
cat daemonset.yaml
---
apiVersion: v1
kind: ConfigMap
metadata:
  name: docker-nanny-install
  namespace: kube-system
data:
  docker-nanny-install.sh: |
    #!/usr/bin/env bash
    /usr/sbin/chroot /host /bin/bash -c "/bin/systemctl enable docker-healthcheck.service"
    /usr/sbin/chroot /host /bin/bash -c "/bin/systemctl start  docker-healthcheck.service"
---
apiVersion: extensions/v1beta1
kind: DaemonSet
metadata:
  name: docker-nanny
  namespace: kube-system
  labels:
    app: docker-nanny
spec:
  template:
    metadata:
      name: docker-nanny
      labels:
        app: docker-nanny
    spec:
      hostNetwork: true
      hostPID: true
      hostIPC: true
      containers:
      - name: systemctl
        image: docker.io/ubuntu:16.04
        command:
        - /usr/local/bin/docker-nanny-install.sh
        imagePullPolicy: IfNotPresent
        securityContext:
          privileged: true
        volumeMounts:
        - name: dev
          mountPath: /host/dev
        - name: proc
          mountPath: /host/proc
        - name: sys
          mountPath: /host/sys
        - name: rootfs
          mountPath: /host
        - name: script
          mountPath: /usr/local/bin
      volumes:
      - name: proc
        hostPath:
          path: /proc
      - name: dev
        hostPath:
          path: /dev
      - name: sys
        hostPath:
          path: /sys
      - name: rootfs
        hostPath:
          path: /
      - name: script
        configMap:
          name: docker-nanny-install
          items:
            - key: docker-nanny-install.sh
              path: docker-nanny-install.sh
          defaultMode: 200
```

**Comment by chrislovecnm** [comment]
Fixed can anyone test?  You will need to do a `make upload` in order to use a custom nodeup.

**Comment by justinsb** [comment]
This is fixed, but let's close when we do alpha.2

**Comment by everettcaleb** [comment]
Is this fix in the latest (k8s-1.5-debian-jessie-amd64-hvm-ebs-2017-03-07 (ami-1e9a3c08)) AMI? 

If I do systemctl --all | grep docker-healthcheck, I get this:

    docker-healthcheck.service    loaded    inactive dead      Run docker-healthcheck once

**Comment by robinmonjo** [comment]
Described as fixed in the `1.6.0-alpha.2` release, closing :+1:
